project/cmake/CMakeLists.txt

```diff
@@ -199,14 +205,22 @@ endif()
 if(NOT WIN32)
   core_optional_dep(OpenGl)
   if(OPENGL_FOUND)
-    core_optional_dep(X ENABLE_X11)
-    core_optional_dep(LibDRM ENABLE_X11)
-    core_optional_dep(XRandR ENABLE_X11)
+    if (ENABLE_MIR)
+      core_optional_dep(Mir ENABLE_MIR)
```

Comment:
For the option on line 58, how would we define a system is mir and not x11? As any system thats linux will most likely be for an x11 system (that possibly supports mir)